lib/pure/math.nim

```diff
@@ -914,6 +916,23 @@ when not defined(js):
       ##  echo log2(0.0)  # -inf
       ##  echo log2(-2.0) # nan
 
+  proc copySign*[T: float32 | float64](x, y: T): T {.importc: "copysign", header: "math.h".} =
+    ## Returns a value with the magnitude of `x` and the sign of `y`.
+    runnableExamples:
+      doAssert copysign(10.0, -1.0) == -10.0
+      doAssert copysign(-10.0, -1.0) == -10.0
+      doAssert copysign(-10.0, 1.0) == 10.0
+      doAssert copySign(10'f32, -1.0) == -10.0
+
+      doAssert copySign(Inf, -1.0) == -Inf
+      doAssert copySign(-Inf, 1.0) == Inf
+      doAssert copySign(1.0, -0.0) == -1.0
+      doAssert copySign(0.0, -0.0) == -0.0
+      doAssert copySign(-1.0, 0.0) == 1.0
+      doAssert copySign(10.0, 0.0) == 10.0
+
+    # TODO add examples for isNaN and JS version
```

Comment:
and VM; which is why might be preferable to have auxiliary importc proc, as was for with `math.isNaN` see https://github.com/nim-lang/Nim/pull/16179